### ISSUE TITLE
Madgwick filtered orientation does not change

### ISSUE THREAD

**Issue by SahanGura** [CCNYRoboticsLab/imu_tools#145]
I've met with an issue of not updating the filtered orientation from the madgwick filter. When I echo the /imu/data & /imu/rpy/filtered ros topics, it shows data publishing to these topics; but orientation stays in a fixed value. In addition, /imu/data_raw and /imu/rpy/raw topics get different values from the imu. This basically means, even though the imu gives correct raw values, filter only calculates an initial value. After that filter seems to be not calculcating orientation. (ROS melodic with Ubuntu 18.04)

Following output shows two consecutive ros messages published to /imu/data topic

header: 
  seq: 159400
  stamp: 
    secs: 0
    nsecs:         0
  frame_id: "map"
orientation: 
  x: -0.0382443172193
  y: -0.0175256838826
  z: -0.0250460517673
  w: 0.998800739835
orientation_covariance: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
angular_velocity: 
  x: -0.00224431836978
  y: -0.00180973904207
  z: 0.00212689721957
angular_velocity_covariance: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
linear_acceleration: 
  x: 0.455426901579
  y: -0.75427287817
  z: 9.84134292603
linear_acceleration_covariance: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
---
header: 
  seq: 159401
  stamp: 
    secs: 0
    nsecs:         0
  frame_id: "map"
orientation: 
  x: -0.0382443172193
  y: -0.0175256838826
  z: -0.0250460517673
  w: 0.998800739835
orientation_covariance: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
angular_velocity: 
  x: 0.00707674538717
  y: -0.00926658976823
  z: -0.00266679259948
angular_velocity_covariance: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
linear_acceleration: 
  x: 0.462618678808
  y: -0.737498760223
  z: 9.65930557251
linear_acceleration_covariance: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
---

**Comment by SahanGura** [comment]
I could solve this issue. I was working in the default branch while working with ros melodic. Once I switched to the melodic branch this issue could be solved. I have not added timestamps to the data_raw messages so it gives me an error when I was working in the melodic branch. After I added timestamps to the messages, madgwick filter started working fine. However, complimentary filter worked without any issue without timestamps
